memory/src/lib.rs

```diff
@@ -116,14 +120,20 @@ where
         RowMajorMatrix::new(rows.concat(), NUM_MEM_COLS)
     }
 
-    fn generate_permutation_trace(
-        &self,
-        _machine: &M,
-        main_trace: RowMajorMatrix<Self::F>,
-        random_elements: Vec<Self::FE>,
-    ) -> RowMajorMatrix<Self::F> {
-        LogUp::<NUM_MEM_LOOKUPS, LOOKUP_DEGREE_BOUND>::new(MEM_LOOKUPS)
-            .build_trace(&main_trace, random_elements)
+    fn global_receives(&self, machine: &M) -> Vec<Interaction<M::F>> {
+        let is_read: VirtualPairCol<M::F> = VirtualPairCol::single_main(MEM_COL_MAP.is_read);
+        let addr = VirtualPairCol::single_main(MEM_COL_MAP.addr);
+        let value = MEM_COL_MAP.value.0.map(VirtualPairCol::single_main);
+
+        let mut fields = vec![is_read, addr];
+        fields.extend(value);
+
+        let receive = Interaction {
+            fields: vec![],
```

Comment:
Just checking, this should be the fields vector defined above?

Reply:
Oops good catch! Forgot to change IDE defaults.